### ISSUE TITLE
daemonset pod fails to come up with HostPort conflict, doesn't retry

### ISSUE THREAD

**Issue by macb** [kubernetes/kubernetes#22716]
I have a DaemonSet that looks like:

```yaml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: loadbalancer
  labels:
    app: loadbalancer
spec:
  template:
    metadata:
      labels:
        app: loadbalancer
    spec:
      nodeSelector:
        type: loadbalancer
      containers:
      - image: "registry.com/macb/haproxy:abc123"
        imagePullPolicy: Always
        name: haproxy
        ports:
        - containerPort: 80
          hostPort: 80
      - image: prom/haproxy-exporter
        imagePullPolicy: Always
        name: exporter
        ports:
        - containerPort: 9101
        args:
        - '-haproxy.scrape-uri=http://user:pass@127.0.0.1:3000/?stats;csv'
      - image: panoptix/rsyslog
        imagePullPolicy: Always
        name: syslog
        ports:
        - containerPort: 514
```

And then multiple nodes that match this daemonset requirement:
```
NAME                                             LABELS                                                                                                    STATUS    AGE
kubelet01  kubernetes.io/hostname=kubelet01,prometheus=true,type=loadbalancer   Ready     13d
kubelet02  kubernetes.io/hostname=kubelet02,type=loadbalancer                   Ready     48d
```

However, only 1 of the 2 nodes are running the expected pod.
```
$ kubectl get po -l app=loadbalancer --show-all
NAME                 READY     STATUS             RESTARTS   AGE
loadbalancer-bgk7i   0/3       HostPortConflict   0          3h
loadbalancer-snmen   3/3       Running            1          4h
```

With the daemonset showing 1 failed pod:
```
$ kubectl describe ds loadbalancer
Name:           loadbalancer
Image(s):       registry.com/macb/haproxy:abc123,prom/haproxy-exporter,panoptix/rsyslog
Selector:       app=loadbalancer
Node-Selector:  type=loadbalancer
Labels:         app=loadbalancer
Desired Number of Nodes Scheduled: 2
Current Number of Nodes Scheduled: 2
Number of Nodes Misscheduled: 0
Pods Status:    1 Running / 0 Waiting / 0 Succeeded / 1 Failed
No events.
```

However, on the node that is not running the pod there is nothing on the required host port of `80`.
```
$ sudo netstat -tulpn | grep 80
$
```

And none of the other scheduled pods appear to be using `hostPort` of any kind.

**Comment by ArtfulCoder** [comment]
cc @mikedanese

**Comment by mikedanese** [comment]
What version of kubernetes are you using. This sounds like https://github.com/kubernetes/kubernetes/issues/22309 which is fixed on HEAD

**Comment by macb** [comment]
kubernetes 1.1.3

When upgrading a cluster with a similar DS to 1.2alpha we ran into #22309 but in that case all DS were offline whereas now it's inconsistent if it will be offline. Manually deleting the failed pod results in a new pod being scheduled that comes up successfully it seems.

**Comment by macb** [comment]
Seems this was fixed in 1.2.
